framework/packages/abstract-interface/src/native/ans_host.rs

```diff
@@ -46,206 +50,137 @@ where
     }
 }
 
-/// Implementation for the daemon, which maintains actual state
-#[cfg(feature = "daemon")]
-use cw_orch::prelude::Daemon;
-
-#[cfg(feature = "daemon")]
-impl AnsHost<Daemon> {
-    pub fn update_all_local(&self) -> Result<(), crate::AbstractInterfaceError> {
-        self.update_assets_local()?;
-        self.update_contracts_local()?;
-        self.update_pools_local()?;
-        Ok(())
-    }
+pub trait Resolver<Chain: CwEnv> {
```

Comment:
`Resolver` is an active statement. To implement `Resolver` for an `Entry` is like saying the `Entry` is the `Resolver`, which is not the case. 

Maybe we can rename it to `ClientResolve`?

Reply:
Also because the `Resolve` and `ClientResolve` are so similar it might be nice to try and merge the traits. 

I'm not entirely sure how you would go about doing that but you could try to move the AnsHost functions to a trait. Then implement that trait for the interface and the abstract-core object and then use the trait as a trait bound. 

```rust
trait <T: Resolver> Resolve<T> {
    type Output;

    fn resolve(&self, resolver: &T) -> Self::Output;
}
```
Then implement it with the resolver trait.

```rust
impl <T: Resolver> Resolve<T> for AssetEntry {
    // ...
    fn resolve(&self, resolver: &T) -> Self::Output {
        resolver.query_asset_entry(self)
    }
}
```

Maybe there's a more elegant way?

Reply:
If we okay to make breaking change of resolve trait on contracts, because currently it takes two params: `QuerierWrapper` and `AnsHost` address. But for suggested change most suitable would be to use `AbstractNameServiceClient` as `Resolver`?